src/server/game/Maps/Map.cpp

```diff
@@ -678,6 +678,31 @@ void Map::Update(const uint32 t_diff)
         player->Update(t_diff);
 
         VisitNearbyCellsOf(player, grid_object_update, world_object_update);
+
+        // Handle updates for creatures in combat with player and are more than 60 yards away
+        if (player->IsInCombat())
+        {
+            std::vector <Creature*> updateList;
+            HostileReference* ref = player->getHostileRefManager().getFirst();
+
+            while (ref)
+            {
+                if (Unit* unit = ref->GetSource()->GetOwner())
+                    if (unit->GetTypeId() == TYPEID_UNIT && unit->ToCreature() && unit->GetMapId() == player->GetMapId() && !unit->IsWithinDistInMap(player, GetVisibilityRange(), false))
+                        updateList.push_back(unit->ToCreature());
+
+                ref = ref->next();
+            }
+
+            // Process deferred update list for player
+            if (!updateList.empty())
+            {
+                for (uint32 i = 0; i<updateList.size(); i++)
+                    VisitNearbyCellsOf(updateList[i], grid_object_update, world_object_update);
+
+                updateList.clear();
```

Comment:
Yep. I wasn't sure if there were memory implications of letting it go out of scope with data in. But I read since (when comparing the various types discussed) that they have a proper destructor anyway that will take care of it.